app/scripts/controllers/transactions.js

```diff
@@ -249,6 +258,21 @@ module.exports = class TransactionController extends EventEmitter {
     })
   }
 
+  resendTransactionAsDuplicate (txMeta) {
+    return new Promise((resolve, reject) => {
+      this.addUnapprovedTransaction(txMeta.txParams, (err, newTxMeta) => {
+        if (err) {
+          delete txMeta.ignore
+          this.updateTx(txMeta)
+          reject(err)
```

Comment:
should we return here or do you intend to keep going and set the ignore to true and resolve the tx